mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulPlayScreen.kt

```diff
@@ -3,32 +3,67 @@ package ch.epfl.sdp.mobile.state
 import androidx.compose.foundation.layout.PaddingValues
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.chess.ChessFacade
+import ch.epfl.sdp.mobile.application.chess.Match
 import ch.epfl.sdp.mobile.ui.play.PlayScreen
 import ch.epfl.sdp.mobile.ui.play.PlayScreenState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
 /**
- * Player class that holds info about chess player
- * @param newGame Callable when new game button is actioned. Switches to pregame screen
+ * Class that holds the necessary info of the [PlayScreen]
+ * @param user the current [AuthenticatedUser]
+ * @param localGame Callable when new game button is actioned for local games. Creates a local game
+ * and switches to it.
+ * @param onlineGame Callable when new game button is actioned for online games. Switches to
+ * [PrepareGameScreen]
+ * @param chessFacade The chess facade to act on the store
+ * @param scope The [Composable]'s [CoroutineScope]
  */
-/* FIXME: add more useful args : user... */
-class Player(private val newGame: () -> Unit) : PlayScreenState {
-  /* TODO: override other attributes */
-  override val onNewGameClick = newGame
+class SnapshotPlayScreenState(
+    override val user: AuthenticatedUser,
+    private val localGame: (match: Match) -> Unit,
+    private val onlineGame: () -> Unit,
+    private val chessFacade: ChessFacade,
+    val scope: CoroutineScope,
```

Comment:
This must be private

Reply:
Fixed in [5e571d3](https://github.com/epfl-SDP/android/pull/226/commits/5e571d34ce190a94d41daf7475bd22d932fe91fe)